oasislmf/utils/data.py

```diff
@@ -360,6 +368,28 @@ def get_dataframe(
     return df
 
 
+def get_dtypes_and_required_cols(get_dtypes):
+    """
+    Get OED column data types and required column names from JSON.
+
+    :param get_dtypes: method to get dict from JSON
+    :type get_dtypes: function
+    """
+    dtypes = get_dtypes()
+    col_dtypes = {
```

Comment:
This only deals with string columns - there are of course OED columns with SQL data types which map to other Python data types such as ints and floats, so the name `col_dtypes` is not quite accurate, because you're only interested in columns whose (Python) values are expected to be strings.

Also, as you're only selecting string columns and set the values for the keys to `v['py_dtype']`, which by construction must be `str`, every value in this dictionary will be `str`. I don't think is useful or necessary.